src/scripts/main.js

```diff
@@ -1 +1,74 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  let resolved = false;
+
+  const timeout = setTimeout(() => {
+    if (!resolved) {
+      reject(new Error('First promise was rejected'));
+    }
+  }, 3000);
+
+  document.addEventListener('click', (e) => {
+    if (e.button === 0 && !resolved) {
+      clearTimeout(timeout);
+      resolved = true;
+      resolve('First promise was resolved');
+    }
+  });
+});
+
+const secondPromise = new Promise((resolve) => {
+  const onClick = (e) => {
+    if (e.button === 0 || e.button === 2) {
+      e.preventDefault();
+      resolve('Second promise was resolved');
+      document.removeEventListener('click', onClick);
+      document.removeEventListener('contextmenu', onClick);
+    }
+  };
+
+  document.addEventListener('click', onClick);
+  document.addEventListener('contextmenu', onClick);
+});
```

Comment:
This promise is resolved when the user clicks the left or right mouse button. But it's not necessary to remove the event listeners after the promise is resolved. Because if the promise is resolved, it will not be able to be resolved again. The Promise object will ignore any other attempts to resolve after the first call. So, you can remove the calls `document.removeEventListener('click', onClick)` and `document.removeEventListener('contextmenu', onClick)`.